api/jobs/gears.py

```diff
@@ -111,7 +111,7 @@ def validate_gear_config(gear, config_):
         validator = Draft4Validator(ci)
 
         try:
-            validator.validate(config_)
+            validator.validate(fill_gear_default_values(gear, config_))
```

Comment:
Defaults will be filled when the job is spawned, so I would think we wouldn't fill them here. What is the expected behavior if a gear is updated and the update has a new default that the rule didn't explicitly set? If we fill defaults here, it won't use the updated default. It is possible they wouldn't want to use an updated default without any user action. 

If I missed a discussion on this, let me know. @kofalt @ambrussimon

Reply:
Given that it's a job rule, I would expect valid defaults to be used, rather than invalid ones from a different gear document. Really the core issue here is that `fill_gear_default_values` modifies the map, it's reasonable to call this in places that you won't want to change.

Reply:
So I would either not call `fill_gear_default_values` in places where you don't want to modify the document (quick fix), or (better fix) change `fill_gear_default_values` to use `copy.deepcopy`, then check the various callees.

Reply:
That's a good point, but I was asking if we wanted defaults persisted to the database for rule config if the user didn't explicitly send them.